vrc-get-gui/src/os_linux.rs

```diff
@@ -69,3 +76,107 @@ fn os_release() -> Option<String> {
         _ => None,
     }
 }
+
+static APPDIR: ArcSwapOption<OsString> = ArcSwapOption::const_empty();
+
+pub fn initialize(app_handle: tauri::AppHandle) {
+    APPDIR.store(app_handle.env().appdir.clone().map(Arc::from));
+}
+
+pub fn open_that(path: impl AsRef<OsStr>) -> io::Result<()> {
+    // We implement open_that here since we have to fix env variables for xdg-open.
+    // This implementation supports xdg-open only but it's general enough for most cases, I think
+
+    let mut command = Command::new("xdg-open");
+    command.arg(path);
+    fix_env_variables(&mut command);
+    let status = command
+        .stdin(Stdio::null())
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .status()?;
+
+    if !status.success() {
+        return Err(io::Error::new(
+            io::ErrorKind::Other,
+            format!("Launcher xdg-open failed with {:?}", status),
+        ));
+    }
+
+    Ok(())
+}
+
+pub(super) fn fix_env_variables(command: &mut Command) {
+    // NOTE: this does not handle env_clear correctly
+
+    let appdir = APPDIR.load();
+    let Some(appdir) = appdir.as_ref() else {
+        return;
+    };
+    let appdir = appdir.as_ref();
+    let appdir_bytes = appdir.as_bytes();
+
+    if appdir_bytes.contains(&b':') {
```

Comment:
```suggestion
    if appdir_bytes.contains(&b':') {
```

appdir_bytes.is_empty()